File: src/Java/SMIL/SMIL.java
import java.util.Scanner;

public class SMIL {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    String input = sc.nextLine();
    String smile1 = ":)";
    String smile2 = ";)";
    String smile3 = ":-)";
    String smile4 = ";-)";
    for (int i = 0; (i = input.indexOf(smile1, i)) != -1; i++) {
      System.out.println(i);
    }
    for (int i = 0; (i = input.indexOf(smile2, i)) != -1; i++) {
      System.out.println(i);
    }
    for (int i = 0; (i = input.indexOf(smile3, i)) != -1; i++) {
      System.out.println(i);
    }
    for (int i = 0; (i = input.indexOf(smile4, i)) != -1; i++) {
      System.out.println(i);
    }
    sc.close();
  }
}

File: src/JavaScript/SMIL/SMIL.js
const readline = require('readline');
const rl = readline.createInterface(process.stdin);

rl.on('line', line => {
  var input = line;
  var smile1 = /:\)/g, res1;
  var smile2 = /;\)/g, res2;
  var smile3 = /:-\)/g, res3;
  var smile4 = /;-\)/g, res4;

  while (res1 = smile1.exec(input)) {
    console.log(res1.index);
  }

  while (res2 = smile2.exec(input)) {
    console.log(res2.index);
  }

  while (res3 = smile3.exec(input)) {
    console.log(res3.index);
  }

  while (res4 = smile4.exec(input)) {
    console.log(res4.index);
  }

  rl.close();
})